src/passes/ConstantFieldPropagation.cpp

```diff
@@ -563,7 +557,7 @@ struct ConstantFieldPropagation : public Pass {
     // a copy of A means it could be a copy of B or C).
     StructUtils::TypeHierarchyPropagator<StructUtils::CombinableBool>
       boolPropagator(subTypes);
-    boolPropagator.propagateToSubTypes(combinedCopyInfos);
+    boolPropagator.propagateToSubTypesWithExact(combinedCopyInfos);
```

Comment:
The comment above explains in detail why we propagate as we do, and should be updated for exactness.

Perhaps something like "when we read from an exact reference, the value at its field may be influenced by writes to non-exact super- and subtypes."

Reply:
Let's address this in follow-on PRs that remove the false distinction between values written by news and values written by sets. The explanation of the analysis can be made much simpler and more rigorous once we simplify the analysis itself.